samples/ChangefeedMigrationSample/Program.cs

```diff
@@ -0,0 +1,193 @@
+            }
+
+            public Task CloseAsync(ChangeFeedObserverContext context, ChangeFeedObserverCloseReason reason)
+            {
+                Console.WriteLine("Worker closed, {0}", context.PartitionKeyRangeId);
+                return Task.CompletedTask;
+            }
+
+            public Task ProcessChangesAsync(ChangeFeedObserverContext context, IReadOnlyList<Document> docs)
+            {
+                DocumentClient newClient = new DocumentClient(new Uri(EndPointUrl), PrimaryKey);
+
+                newClient.CreateDatabaseIfNotExistsAsync(new Database { Id = DBName });
+
+                // create dest collection if it does not exist
+                DocumentCollection destCollection = new DocumentCollection();
+                destCollection.Id = DestCollName;
+                destCollection.PartitionKey.Paths.Add(DestCollPartKey);
+
+                newClient.CreateDocumentCollectionIfNotExistsAsync(
+                    UriFactory.CreateDatabaseUri(DBName),
+                    destCollection,
+                    new RequestOptions { OfferThroughput = 2500 });
+
+                Console.WriteLine("Change feed: total {0} doc(s)", Interlocked.Add(ref s_totalDocs, docs.Count));
+                foreach (Document doc in docs)
+                {
+                    Console.WriteLine(doc.ToString());
+                    newClient.UpsertDocumentAsync(
+                        UriFactory.CreateDocumentCollectionUri(DBName, DestCollName),
+                        doc);
+                }
+                return Task.CompletedTask;
+            }
+        }
+
+        private static async Task UpdateDb(string endpoint, string authKey)
+        {
+            // Use this function to update data in monitored collection in seperate thread
+            // Returns all documents in the collection.
+            Console.WriteLine("Connect client");
+            DocumentClient client = new DocumentClient(new Uri(endpoint), authKey);
+            Uri collectionUri = UriFactory.CreateDocumentCollectionUri(DBName, MonitoredCollectionName);
+
+            Console.WriteLine("Connect database");
+            try
+            {
+                Database database = await client.ReadDatabaseAsync(UriFactory.CreateDatabaseUri(DBName));
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("Connect database failed");
+                Console.WriteLine(e.Message);
+            }
+
+            // Create new documents 
+            System.Console.WriteLine("Create JSON document");
+            for (int i = 0; i < 10; i++)
+            {
+                System.Console.WriteLine("Creating document XMS-0004 {0}", i);
+                await client.UpsertDocumentAsync(
+                UriFactory.CreateDocumentCollectionUri(DBName, MonitoredCollectionName),
+                new DeviceReading
+                {
+                    Id = String.Join("XMS-005-FE24C_", i.ToString()),
```

Comment:
If this example is run twice in a row will it work?  Likely it will have Id collisions.  One fix is to upsert everything or better have the class when its created make a random prefix to use for all data in the example.